jerry-core/ecma/operations/ecma-objects.h

```diff
@@ -27,6 +27,35 @@
  * @{
  */
 
+#if JERRY_ERROR_MESSAGES
+/**
+ * Reject with TypeError depending on 'is_throw' with the given format
+ */
+#define ECMA_REJECT_WITH_FORMAT(is_throw, msg, ...) \
+  ((is_throw) ? ecma_raise_standard_error_with_format (ECMA_ERROR_TYPE, (msg), __VA_ARGS__) : ECMA_VALUE_FALSE)
+
+/**
+ * Reject with TypeError depending on 'is_throw' with the given message
+ */
+#define ECMA_REJECT(is_throw, msg) \
+  ((is_throw) ? ecma_raise_type_error (msg) : ECMA_VALUE_FALSE)
+#else /* !JERRY_ERROR_MESSAGES */
+/**
+ * Reject with TypeError depending on is_throw flags wit the given format
+ */
+#define ECMA_REJECT_WITH_FORMAT(is_throw, msg, ...) \
+  ECMA_REJECT((is_throw), (msg))
+
+/**
+ * Reject with TypeError depending on is_throw flags wit the given message
+ */
+#define ECMA_REJECT(is_throw, msg) \
+  ((is_throw) ? ecma_raise_type_error (NULL) : ECMA_VALUE_FALSE)
+#endif /* JERRY_ERROR_MESSAGES */
+
+ecma_value_t ecma_raise_property_redefinition (ecma_string_t *property_name_p, uint16_t flags);
+ecma_value_t ecma_raise_readonly_assignment (ecma_string_t *property_name_p, bool is_throw);
```

Comment:
To make it consistent should we also pass `flags` here? (just like in the `ecma_raise_property_redefinition` method)

Reply:
Currently `ecma_raise_readonly_assignment ` is only used by `[[Put]]` method where the `is_throw` is passed as a boolean. we should use a ternary to convert it to flags, so I'd leave as it for now.

Reply:
True, ok. We'll take a look at it later if we really want to change it.